### PR TITLE
Allow to reference a test repo in test job metadata

### PR DESCRIPTION
**Pull request by mmuzila**
This change allows to specify distinct repository for tests.

Fixes: #968

### CODE REVIEW

File: packit_service/worker/testing_farm.py
Comment (jpopelka) [inline packit_service/worker/testing_farm.py:68]: Why do we check for `fmf_url` and not `fmf_ref`?

**Review by jpopelka — COMMENTED** [review]
LGTM with one question.

Would be nice to finish packit/packit#1155 so that tests here can pass.

**Comment by mmuzila** [comment]
> LGTM with one question.
> 
> Would be nice to finish [packit/packit#1155](https://github.com/packit/packit/pull/1155) so that tests here can pass.

I want to merge it after release of  [packit/packit#1155](https://github.com/packit/packit/pull/1155). I probably should use WIP prefix for this PR.

File: packit_service/worker/testing_farm.py
Comment (mmuzila) [reply packit_service/worker/testing_farm.py:68]: There is no need to check for `fmf_ref`. If it isn't set, `None` is returned and it shouldn't be a problem.

I check for `fmf_url` because I want to use either (project repo + project ref) or (remote repo + remote ref). I don't want to mix them. 

This way if `fmf_ref` is set, but `fmf_url` isn't, tests are taken from project repository and current branch/ref (`fmf_ref` is ignored).

If those parameters were independent it would lead to problems. Eg. if `fmf_url` was set, `fmf_ref` wasn't set and project ref was used instead, testing farm would receive the remote repo url (`fmf_url`) with invalid ref (project repo commit-sha)

I agree that this may be a bit confusing.

File: packit_service/worker/testing_farm.py
Comment (jpopelka) [inline packit_service/worker/testing_farm.py:85]: Per [TF docs](https://testing-farm.gitlab.io/api/#operation/requestsPost), the "ref" is optional so if `fmf_ref` is not set (and `fmf_url` is set), we should not send it at all, rather than sending `{"ref": None}`.

**Review by TomasTomecek — APPROVED** [review]
LGTM, though I'd wait for Jirka's approval as well

**Comment by TomasTomecek** [comment]
recheck

**Review by jpopelka — APPROVED** [review]

**Comment by TomasTomecek** [comment]
recheck

**Comment by jpopelka** [comment]
recheck

**Comment by jpopelka** [comment]
>   * [packit-service-tests] : FAILURE in 2m 23s

It's still failing because we are now (since #977) using pre-built tests image instead of building it each time.
The pros are speed (over 2min instead of over 8min), the cons are that if you need to rebuild it because something has changed in packit since last push to this repo, you have to do it yourself in [Actions](https://github.com/packit/packit-service/actions), which is what I did just now.

**Pull request merged**
